lib/index.js

```diff
@@ -121,36 +102,27 @@ function setupRateLimitEventExt(server, options) {
     const currentRlRequest = pendingRequests.get(request.id);
     if (currentRlRequest && !currentRlRequest.isConformant()) { return; }
 
-    const limitd = limitdPool.get(options.address);
     getType(request, reply, (err, type) => {
-      extractKeyAndTakeToken(limitd, request, reply, type);
+      extractKeyAndTakeToken(options.limitd, request, reply, type);
     });
   });
 }
 
 function setupStopExt(server, options) {
   server.ext('onPostStop', function(server, next) {
-    const limitd = limitdPool.get(options.address);
+    const limitd = options.limitd;
     if (limitd) {
       limitd.disconnect();
```

Comment:
Since now the limitd instance doesn't belong to Patova we should not disconnect here.